Model/Media.php

```diff
@@ -643,8 +643,11 @@ public function getCategory()
     /**
      * @param CategoryInterface $category|null
      */
-    public function setCategory(CategoryInterface $category = null)
+    public function setCategory($category = null)
```

Comment:
For those like me who think this does not make sense, [turns out it indeed does not](https://bugs.php.net/bug.php?id=69612). I can't find a solution, apart from using a conditional class declaration : 

```php
if (somecondition) {
    class Media {
        //version 1
    }
} else {
    class Media {
        //version 2
    }
}
```

This would be very ugly. @vguardiola , I think you should do the PR on the master branch, and drop the backwards compatibility layer.